odl/space/space_utils.py

```diff
@@ -22,14 +22,27 @@
 from future import standard_library
 standard_library.install_aliases()
 
-__all__ = ('vector',)
+__all__ = ('vector', 'ntuples', 'fn', 'cn', 'rn',
+           'NTUPLES_IMPLS', 'FN_IMPLS')
 
 import numpy as np
 
-from odl.space.ntuples import Rn, Cn, Fn, Ntuples
+from odl.set import RealNumbers, ComplexNumbers
+from odl.space.numpy_ntuples import NumpyNtuples, NumpyFn
 from odl.util.utility import (
     is_real_floating_dtype, is_complex_floating_dtype, is_scalar_dtype)
-from odl.space.cu_ntuples import CUDA_AVAILABLE, CudaRn, CudaFn, CudaNtuples
+from pkg_resources import iter_entry_points
+
+
+NTUPLES_IMPLS = {'numpy': NumpyNtuples}
+FN_IMPLS = {'numpy': NumpyFn}
+for entry_point in iter_entry_points(group='odl.space', name=None):
+    try:
+        module = entry_point.load()
+        NTUPLES_IMPLS.update(module.ntuples_impls())
+        FN_IMPLS.update(module.fn_impls())
+    except ImportError:
```

Comment:
Yes the library is imported somehow inside the "load" call. And yes, it will fail if the submodule lacks a library or similarily (basically triggering an importerror). I added this to make sure that a faulty pluggin does not render a odl installation unusable.

Regarding doing a small test Ill have to disagree. We used to do this a long time ago, but it came with a notable performance penalty (import odl became about 3 times slower, and were already slower than numpy) and honestly only helped hiding bugs.

If a plugin is installed, we should i.m.o run the tests for it. That they fail is in this case correct and indicates an actual bug.

What we could do is add some exclude flag: "--exclude-plugins=cuda" to the pytest config. Then users can opt to skip them, if the user wants to.